common/src/sql/dbinit.sql

```diff
@@ -223,6 +220,60 @@ CREATE INDEX ON omicron.public.silo_user (
 ) WHERE
     time_deleted IS NULL;
 
+CREATE TYPE omicron.public.provider_type AS ENUM (
+  'local',
+  'saml',
+  'ldap'
+);
+
+/*
+ * Silo identity provider list
+ */
+CREATE TABLE omicron.public.silo_identity_provider (
+    silo_id UUID NOT NULL,
+    provider_type omicron.public.provider_type NOT NULL,
+    name TEXT NOT NULL,
+    provider_id UUID NOT NULL,
+
+    PRIMARY KEY (silo_id, provider_id)
+);
+
+CREATE INDEX ON omicron.public.silo_identity_provider (
+    silo_id,
+    provider_id
+);
+
+/*
+ * Silo SAML identity provider
+ */
+CREATE TABLE omicron.public.silo_saml_identity_provider (
+    /* Identity metadata */
+    id UUID PRIMARY KEY,
+    name STRING(128) NOT NULL,
+    description STRING(512) NOT NULL,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL,
+    time_deleted TIMESTAMPTZ,
+
+    silo_id UUID NOT NULL,
+
+    idp_metadata_url TEXT NOT NULL,
+    idp_metadata_document_string TEXT NOT NULL,
+
+    idp_entity_id TEXT NOT NULL,
+    sp_client_id TEXT NOT NULL,
+    acs_url TEXT NOT NULL,
+    slo_url TEXT NOT NULL,
+    technical_contact_email TEXT NOT NULL,
+
+    public_cert TEXT,
+    private_key TEXT
+);
+
+CREATE INDEX ON omicron.public.silo_saml_identity_provider (
```

Comment:
done in 2ef7da6e, though I didn't add the id column. I was having issues because the delete from this table wasn't implemented as a soft delete, which is also fixed in 2ef7da6e.

Reply:
I think the `id` column is necessary.  Without it, you won't be able to have two SAML identity providers for a Silo, and you also won't be able to paginate the list of SAML identity providers for a Silo.  (We could decide to forego that here if we want -- we're presumably not going to have zillions of these -- it just seemed easier to me to be consistent with everything else.)